code/game/mecha/equipment/tools/mining_tools.dm

```diff
@@ -129,7 +129,7 @@
 		target.apply_damage(10, BRUTE, BODY_ZONE_CHEST, target.run_armor_check(target_part, MELEE))
 
 		//blood splatters and sparks
-		if(issilicon(target)  || isbot(target) || isswarmer(target) || !IS_ORGANIC_LIMB(target_part))
+		if(issilicon(target)  || isbot(target) || isswarmer(target) || IS_ROBOTIC_LIMB(target_part))
```

Comment:
`!` doesn't work here because this is supposed to be robotic only but it will cause sparks just because your limb is not organic

Reply:
a limb must either be robot or organic. ~the author of this code

Reply:
I (in)famously authored bodypart code. Organ/Robot are mutually exclusive. You cannot have both, and you MUST have one.

Reply:
I know you are the author of limb system.
I know all non-organic limbs are technically robotic because there is only organic v.s. robotic in the current design
I know you are not a fan of having another macro that checks if a limb is robotic when there is only organic v.s. robotic.

I am saying this line is an example of a possibly incorrect behaviour when someone wants to do a behaviour that's specifically dedicated for robotic things with the said examples.
I am more interested to hear what you are expecting from me to do instead of trying to change my mind. I can follow what you generally want it as is expected.

Reply:
Testing for robotic limbs makes more sense here. What if we want to add stone limbs that aren't robotic but also aren't organic. Then only limbs with some robotic flag on them should be sparked.